src/Microsoft.Data.SqlClient/netfx/src/Microsoft/Data/SqlClient/SqlConnectionString.cs

```diff
@@ -993,6 +996,30 @@ internal SqlConnectionColumnEncryptionSetting ConvertValueToColumnEncryptionSett
             }
         }
 
+        internal SqlConnectionAttestationProtocol ConvertValueToAttestationProtocol()
+        {
+            object value = base.Parsetable[KEY.AttestationProtocol];
+
+            string valStr = value as string;
```

Comment:
I’m just a bit confused, if ‘value’ is null, can it be converted to ‘string’ here? I see you’re checking `valStr is null`, isn’t that a bit late to check?

Reply:
Do we have test cases to set this connection string property as `null`?

Reply:
The Attestation Protocol is never null. If it is not defined in connection string, by default it is 0 (NotDefined) and we have tested that in functional tests.

Reply:
I did a bit of digging. This piece of code is executed before the default is set (in fact that's what line 1006 is doing, setting the default). Anyway as for @cheenamalhotra's question, from [here](https://stackoverflow.com/questions/2099900/difference-between-tostring-and-as-string-in-c-sharp)
```
object value = null;
string valStr = value as string;
```
`valStr` is still null